frontend/app/profile/page.tsx

```diff
@@ -0,0 +1,367 @@
+"use client";
+import { Dialog, Transition } from "@headlessui/react";
+import Image from "next/image";
+import { Fragment, useState } from "react";
+import DP from "./../../public/images/Ashley & Owen.png";
+import eye from "./../../public/images/Eye.png";
+import location from "./../../public/images/location.png";
+import Upload from "./../../public/images/upload.png";
+
+const calculateAge = (birthday) => {
+  const [day, month, year] = birthday.split("-");
+  const birthDate = new Date(`${year}-${month}-${day}`);
+  const currentDate = new Date();
+  let age = currentDate.getFullYear() - birthDate.getFullYear();
+  if (
+    currentDate.getMonth() < birthDate.getMonth() ||
+    (currentDate.getMonth() === birthDate.getMonth() &&
+      currentDate.getDate() < birthDate.getDate())
+  ) {
+    age--;
+  }
+  return age;
+};
+
+const initialProfileData = {
+  firstName: "Reggiestar",
+  lastName: "Yang",
+  location: "Sydney, Australia",
+  phoneNumber: "0468368618",
+  email: "owya490@gmail.com",
```

Comment:
long emails go off the page in mobile view - see if you can try wrapping the email so that it goes to a new line if it is too long?